workspace/popjava/src/popjava/service/jobmanager/search/SNExploration.java

```diff
@@ -0,0 +1,55 @@
+package popjava.service.jobmanager.search;
+
+import java.util.LinkedList;
+import popjava.baseobject.POPAccessPoint;
+import popjava.buffer.POPBuffer;
+import popjava.dataswaper.IPOPBase;
+
+/**
+ * Exploration list on the network in the grid.
+ * Keep track of each node to avoid infinite loops.
+ * @author Davide Mazzoleni
+ */
+public class SNExploration implements IPOPBase {
+	private static final int MAX_QUEUE_SIZE = 500;
+	
+	private final LinkedList<POPAccessPoint> visited = new LinkedList<>();

```

Comment:
Define as List<POPAccessPoint>, not linked list.

Second question, why a linked list, and not an ArrayList? Considering the use-case, an ArrayList should be better